spec/integ/crypto/megolm-backup.spec.ts

```diff
@@ -83,6 +84,58 @@ afterEach(() => {
     indexedDB = new IDBFactory();
 });
 
+enum MockKeyUploadEvent {
+    KeyUploaded = "KeyUploaded",
+}
+
+type MockKeyUploadEventHandlerMap = {
+    [MockKeyUploadEvent.KeyUploaded]: (roomId: string, sessionId: string, backupVersion: string) => void;
+};
+
```

Comment:
I am not a fan of the use of EventEmitter in tests like this. Is there any reason we can't just pass an `onBackupRequest` callback into `mockUploadEmitter` ?

Reply:
Mainly because I found it more easy to then create Promise that resolve when the event is emitted.
I tried the other way but the test was more annoying to write (and read), but I am not very proficient in writing promises

Reply:
mm fair enough. I probably wouldn't do this personally but I'm not sure I can come up with anything significantly better.